napari/_qt/widgets/qt_keyboard_settings.py

```diff
@@ -0,0 +1,255 @@
+from collections import OrderedDict
+
+from qtpy.QtCore import Qt
+from qtpy.QtWidgets import (
+    QComboBox,
+    QDialog,
+    QHBoxLayout,
+    QHeaderView,
+    QLabel,
+    QPushButton,
+    QTableWidget,
+    QTableWidgetItem,
+    QVBoxLayout,
+    QWidget,
+)
+
+from ...layers import Image, Labels, Points, Shapes, Surface, Vectors
+from ...utils.action_manager import action_manager
+from ...utils.settings import SETTINGS
+from ...utils.translations import trans
+
+
+class ShortcutEditor(QDialog):
+    """ """
+
+    ALL_ACTIVE_KEYBINDINGS = trans._('All active key bindings')
+
+    def __init__(
+        self,
+        # viewer,
+        # key_map_handler,
+        parent: QWidget = None,
+        description: str = "",
+        value: dict = None,
+    ):
+
+        super().__init__(parent=parent)
+
+        layers = [
+            Image,
+            Labels,
+            Points,
+            Shapes,
+            Surface,
+            Vectors,
+        ]
+
+        self.key_bindings_strs = OrderedDict()
+
+        # widgets
+        self.layer_combo_box = QComboBox(self)
+        self._label = QLabel(self)
+        self._table = QTableWidget(self)
+
+        # Set up buttons
+        self._restore_button = QPushButton(trans._("Reset All Keybindings"))
+
+        # set up
+        # all actions
+        all_actions = action_manager._actions.copy()
+        self.key_bindings_strs[self.ALL_ACTIVE_KEYBINDINGS] = []
+        for layer in layers:
```

Comment:
You can likely use `action_manager._actions.values()` it will give you actions that are not bounds and action that acre not on layers (like viewer actions).